Dip/Dip/Dip.swift

```diff
@@ -201,30 +233,30 @@ public final class DependencyContainer {
   /// Actually resolve dependency
   private func _resolve<T, F>(tag: Tag? = nil, key: DefinitionKey?, definition: DefinitionOf<T, F>, builder: F->T) -> T {
     
-    return resolvedInstances.resolve {
-      
-      if let previouslyResolved: T = resolvedInstances.previouslyResolved(key, definition: definition) {
-        return previouslyResolved
-      }
-      else {
-        let resolvedInstance = builder(definition.factory)
+    return threadSafe {
```

Comment:
Maybe it will be better to lock only whole `resolve<T, F>(tag tag: Tag? = nil, builder: F->T) throws -> T ` method that calls this one? The same will apply to other methods that will call `_resolve` in future (if there will be any). As for me less locking operations is better.